### DISCUSSION TITLE
Is it safe to add indexes?

### DISCUSSION THREAD

**Discussion by dekiesel** [firefly-iii/firefly-iii#8259] [Firefly III] [answered]
When running a category-report for a whole year the DB never answers (in my case, the container doesn't have a lot of resources).

I got the SQL-statement it runs and after a quick look I saw that it joins `transactions.transaction_journal_id` on `transaction_journals.id`, which are TOP 2 and TOP 5 of my tables (by count of rows). 

There is no index on `transaction_journals.id`. Is it safe to add one?

Thanks!

**Comment by JC5** [comment, marked as answer]
There isn't? There should be. What's your database?

But it should be safe.

**Reply by dekiesel** [reply]
I am using mariadb
![image](https://github.com/firefly-iii/firefly-iii/assets/17333827/dff46728-4c5b-496b-b002-d70e6f005977)

Thanks for the answer! :)

**Reply by dekiesel** [reply]
Ouch, apologies, I need to take that back! My filter rules prevented me from seeing the PRIMARY key.

![image](https://github.com/firefly-iii/firefly-iii/assets/17333827/0204432e-23fb-4422-87fb-16eebf1febd1)
